web/tests/lib/test_stream.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+let stream_id = 1;
+
+const get_stream_id = () => {
+    stream_id += 1 + Math.floor(Math.random() * 10);
+    return stream_id;
+};
+
+const get_random_string = () => Math.floor(Math.random() * 2 ** 54).toString(36);
+
+exports.make_stream = (opts = {}) => {
```

Comment:
Should this be responsible for adding the stream to the `stream_data` module? I guess perhaps the answer might be no, but I'm not sure.

Reply:
Yes, if we want to use this function only when we have to add stream to the module. Although I am not sure if we will ever use this function and will not add the stream to the `stream_data` module, shouldn't we avoid doing that at this moment?

Reply:
I doubt we will use this without adding them to `stream_data`, and I think this could accept a parameter to disable that behavior, but I am OK with tweaking that detail as we try to roll it out.